### ISSUE TITLE
Explore uploading models to Hugging Face

### ISSUE THREAD

**Issue by evgenyrp** [mozilla/firefox-translations-training#804]
It will likely require converting them to HF format/Pytorch, similar to how it's done for the OPUS-MT and HPLT models:

https://huggingface.co/Helsinki-NLP/opus-mt-zh-en
https://huggingface.co/HPLT/translate-sw-en-v1.0-hplt_opus/tree/main

I also found this converter: 
https://github.com/huggingface/transformers/blob/main/src/transformers/models/marian/convert_marian_to_pytorch.py

There might be useful code here as well:
https://github.com/hplt-project/HPLT-MT-Models/tree/main/v1.0/raw_scripts

**Comment by gregtatum** [comment]
Most of the scripts I looked at don't support the Transformer-RNN structure that we use. Plus we'd have to support the `int8shiftAlphaAll` mode, which is only in forked Marian. I have some details in: [Findings for the Marian to ONNX Investigation](https://docs.google.com/document/d/18PAGLEiG2Emplbkm-XLxu5iPqO17g2tTlRPo9M6VAT4/edit#heading=h.btxed7h5b9pj)

**Comment by eu9ene** [comment]
The HPLT models on HF are 300 Mb in size, so they look more like a Teacher model with transformer-base architecture. We can explore how to upload the student models without conversion and whether they will be usable later somehow. Maybe having proper Python bindings can help to integrate it with HF pipelines.

**Comment by eu9ene** [comment]
> Most of the scripts I looked at don't support the Transformer-RNN structure that we use. Plus we'd have to support the `int8shiftAlphaAll` mode, which is only in forked Marian. I have some details in: [Findings for the Marian to ONNX Investigation](https://docs.google.com/document/d/18PAGLEiG2Emplbkm-XLxu5iPqO17g2tTlRPo9M6VAT4/edit#heading=h.btxed7h5b9pj)

Yes, but it's ONNX. I'm sure you can implement everything that's in Marian in Pytorch. It doesn't mean we have to go this way though as it's definitely some work but it would be interesting to explore.

**Comment by marco-c** [comment]
We could start by uploading teacher models.

**Comment by gregtatum** [comment]
> Yes, but it's ONNX

The investigation wasn't just for ONNX, as there are links out to other converters. But yes, I agree that if we can get out to other formats it gives us much more options on how we can run these things.

**Comment by marco-c** [comment]
See also https://github.com/mozilla/firefox-translations-models/issues/179.

**Comment by FricoRico** [comment]
I'm in way over my head, and I'm still trying to understand all of this so forgive me if I say something stupid; I have indeed confirmed that teacher models are able to be exported, but I guess you have confirmed this as well. For the conversion I used the [Transformers conversion tools](https://github.com/huggingface/transformers/blob/main/src/transformers/models/marian/convert_marian_to_pytorch.py).

The problem is indeed that neither `marian-conv` nor `transformers` support converting the transformer-RNN structures. In theory this would be possible correct? Both ONNX and Transformers seem to support RNN models? I guess it's a matter of improving the existing tooling?

Would love to invest some time into this, but like I said. It is quite out of my league at the moment.

**Comment by FricoRico** [comment]
I have requested help from the transformers package team

https://github.com/huggingface/transformers/issues/36651
